WorkspaceProcessor throws exception in remote dev scenario on Kubernetes

### Describe the bug

WorkspaceProcessor throws exception in remote dev scenario on Kubernetes

```
23:17:20,190 INFO  [app] Caused by: java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.getFileSystem()" because "path" is null
23:17:20,190 INFO  [app] 	at java.base/java.nio.file.Files.provider(Files.java:105)
23:17:20,190 INFO  [app] 	at java.base/java.nio.file.Files.readAttributes(Files.java:1854)
23:17:20,190 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220)
23:17:20,191 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:277)
23:17:20,191 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:323)
23:17:20,191 INFO  [app] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2792)
23:17:20,191 INFO  [app] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2870)
23:17:20,191 INFO  [app] 	at io.quarkus.devui.deployment.menu.WorkspaceProcessor.locateWorkspaceItems(WorkspaceProcessor.java:61)
```

`WorkspaceProcessor` was introduced in https://github.com/quarkusio/quarkus/pull/46907 by @phillip-kruger 


Stacktrace of the fail:
```
23:17:12,088 INFO  Running command: kubectl scale deployment/app --replicas=1 -n ts-vpxxpurhyn
23:17:12,156 INFO  kubectl: deployment.apps/app scaled
23:17:16,267 INFO  [app] [app-5986d6466-prddc] Starting the Java application using /opt/jboss/container/java/run/run-java.sh ...
23:17:16,267 INFO  [app] INFO exec -a "java" java -Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager -cp "." -jar /deployments/quarkus-run.jar 
23:17:16,270 INFO  [app] INFO running in /deployments
23:17:20,181 INFO  [app] Press [e] to edit command line args (currently ''), [h] for more options>
23:17:20,181 INFO  [app] 23:17:16,172 JBoss Threads version 3.8.0.Final
23:17:20,181 INFO  [app] 23:17:16,538 Attempting to start live reload endpoint to recover from previous Quarkus startup failure
23:17:20,181 INFO  [app] 23:17:16,844 JBoss Threads version 3.8.0.Final
23:17:20,182 INFO  [app] 23:17:17,015 Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
Error: 0,182 INFO  [app] 	[error]: Build step io.quarkus.devui.deployment.menu.WorkspaceProcessor#locateWorkspaceItems threw an exception: java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.getFileSystem()" because "path" is null
23:17:20,182 INFO  [app] 	at java.base/java.nio.file.Files.provider(Files.java:105)
23:17:20,182 INFO  [app] 	at java.base/java.nio.file.Files.readAttributes(Files.java:1854)
23:17:20,182 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220)
23:17:20,182 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:277)
23:17:20,183 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:323)
23:17:20,183 INFO  [app] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2792)
23:17:20,183 INFO  [app] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2870)
23:17:20,183 INFO  [app] 	at io.quarkus.devui.deployment.menu.WorkspaceProcessor.locateWorkspaceItems(WorkspaceProcessor.java:61)
23:17:20,183 INFO  [app] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
23:17:20,183 INFO  [app] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
23:17:20,183 INFO  [app] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
23:17:20,184 INFO  [app] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
23:17:20,184 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
23:17:20,184 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
23:17:20,184 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
23:17:20,184 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
23:17:20,184 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:1583)
23:17:20,184 INFO  [app] 	at org.jboss.threads.JBossThread.run(JBossThread.java:499)
23:17:20,185 INFO  [app] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:372)
23:17:20,185 INFO  [app] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:289)
23:17:20,185 INFO  [app] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:61)
23:17:20,185 INFO  [app] 	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:89)
23:17:20,185 INFO  [app] 	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:428)
23:17:20,185 INFO  [app] 	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
23:17:20,185 INFO  [app] 	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
23:17:20,185 INFO  [app] 	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
23:17:20,186 INFO  [app] 	at io.quarkus.deployment.mutability.DevModeTask.main(DevModeTask.java:70)
23:17:20,186 INFO  [app] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
23:17:20,186 INFO  [app] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
23:17:20,186 INFO  [app] 	at io.quarkus.bootstrap.runner.DevModeMediator.doStart(DevModeMediator.java:55)
23:17:20,186 INFO  [app] 	at io.quarkus.bootstrap.runner.DevModeMediator.doDevMode(DevModeMediator.java:34)
23:17:20,186 INFO  [app] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:52)
23:17:20,186 INFO  [app] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:36)
23:17:20,186 INFO  [app] Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
Error: 0,187 INFO  [app] 	[error]: Build step io.quarkus.devui.deployment.menu.WorkspaceProcessor#locateWorkspaceItems threw an exception: java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.getFileSystem()" because "path" is null
23:17:20,187 INFO  [app] 	at java.base/java.nio.file.Files.provider(Files.java:105)
23:17:20,187 INFO  [app] 	at java.base/java.nio.file.Files.readAttributes(Files.java:1854)
23:17:20,187 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220)
23:17:20,187 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:277)
23:17:20,187 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:323)
23:17:20,187 INFO  [app] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2792)
23:17:20,188 INFO  [app] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2870)
23:17:20,188 INFO  [app] 	at io.quarkus.devui.deployment.menu.WorkspaceProcessor.locateWorkspaceItems(WorkspaceProcessor.java:61)
23:17:20,188 INFO  [app] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
23:17:20,188 INFO  [app] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
23:17:20,188 INFO  [app] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
23:17:20,188 INFO  [app] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
23:17:20,188 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
23:17:20,188 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
23:17:20,189 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
23:17:20,189 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
23:17:20,189 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:1583)
23:17:20,189 INFO  [app] 	at org.jboss.threads.JBossThread.run(JBossThread.java:499)
23:17:20,189 INFO  [app] 	at io.quarkus.builder.Execution.run(Execution.java:122)
23:17:20,189 INFO  [app] 	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:78)
23:17:20,189 INFO  [app] 	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:161)
23:17:20,190 INFO  [app] 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:368)
23:17:20,190 INFO  [app] 	... 14 more
23:17:20,190 INFO  [app] Caused by: java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.getFileSystem()" because "path" is null
23:17:20,190 INFO  [app] 	at java.base/java.nio.file.Files.provider(Files.java:105)
23:17:20,190 INFO  [app] 	at java.base/java.nio.file.Files.readAttributes(Files.java:1854)
23:17:20,190 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220)
23:17:20,191 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:277)
23:17:20,191 INFO  [app] 	at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:323)
23:17:20,191 INFO  [app] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2792)
23:17:20,191 INFO  [app] 	at java.base/java.nio.file.Files.walkFileTree(Files.java:2870)
23:17:20,191 INFO  [app] 	at io.quarkus.devui.deployment.menu.WorkspaceProcessor.locateWorkspaceItems(WorkspaceProcessor.java:61)
23:17:20,191 INFO  [app] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
23:17:20,191 INFO  [app] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
23:17:20,192 INFO  [app] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
23:17:20,192 INFO  [app] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
23:17:20,192 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
23:17:20,192 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
23:17:20,192 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
23:17:20,192 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
23:17:20,192 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:1583)
23:17:20,192 INFO  [app] 	at org.jboss.threads.JBossThread.run(JBossThread.java:499)
```

### Expected behavior

https://github.com/quarkus-qe/quarkus-test-framework/blob/main/examples/greetings/src/test/java/io/quarkus/qe/KubernetesRemoteDevModeGreetingResourceIT.java to pass

### Actual behavior

https://github.com/quarkus-qe/quarkus-test-framework/blob/main/examples/greetings/src/test/java/io/quarkus/qe/KubernetesRemoteDevModeGreetingResourceIT.java fails

### How to Reproduce?

Setup environment for https://github.com/quarkus-qe/quarkus-test-framework/wiki/7.-Kubernetes

```
git clone https://github.com/quarkus-qe/quarkus-test-framework/
cd quarkus-test-framework
mvn -pl examples/greetings -Pframework,examples,kubernetes clean verify
```

Another approach can be to look into GH Actions logs - https://github.com/quarkus-qe/quarkus-test-framework/actions/runs/14231376902/job/39882572106
Find `KubernetesRemoteDevModeGreetingResourceIT` and go through the lines identified by "Running command:" prefix

GH Action setup: https://github.com/quarkus-qe/quarkus-test-framework/blob/main/.github/workflows/daily.yaml#L104

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

Java 21

### Quarkus version or git rev

Quarkus main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java
@@ -23,6 +23,7 @@ import io.quarkus.deployment.IsDevelopment;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.BuildSteps;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
 import io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem;
 import io.quarkus.devui.deployment.DevUIConfig;
 import io.quarkus.devui.deployment.InternalPageBuildItem;
@@ -44,54 +45,62 @@ public class WorkspaceProcessor {
 
     @BuildStep
     void locateWorkspaceItems(BuildSystemTargetBuildItem buildSystemTarget,
+            LaunchModeBuildItem launchModeBuildItem,
             BuildProducer<WorkspaceBuildItem> workspaceProducer,
             DevUIConfig devUIConfig) {
+
+        if (launchModeBuildItem.isNotLocalDevModeType()) {
+            return;
+        }
 
         Path outputDir = buildSystemTarget.getOutputDirectory();
         Path projectRoot = outputDir.getParent();
 
-        List<WorkspaceBuildItem.WorkspaceItem> workspaceItems = new ArrayList<>();
+        if (projectRoot != null && Files.exists(projectRoot)) {
 
-        List<String> ignoreFolders = devUIConfig.workspace().ignoreFolders().orElse(new ArrayList<>());
-        ignoreFolders.add("node_modules");
+            List<WorkspaceBuildItem.WorkspaceItem> workspaceItems = new ArrayList<>();
 
-        final List<Pattern> ignoreFilePatterns = devUIConfig.workspace().ignoreFiles().orElse(List.of());
+            List<String> ignoreFolders = devUIConfig.workspace().ignoreFolders().orElse(new ArrayList<>());
+            ignoreFolders.add("node_modules");
 
-        try {
-            Files.walkFileTree(projectRoot, new SimpleFileVisitor<Path>() {
-                @Override
-                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                    if (Files.isHidden(dir) || ignoreFolders.contains(dir.getFileName().toString())) {
-                        return FileVisitResult.SKIP_SUBTREE;
+            final List<Pattern> ignoreFilePatterns = devUIConfig.workspace().ignoreFiles().orElse(List.of());
+
+            try {
+                Files.walkFileTree(projectRoot, new SimpleFileVisitor<Path>() {
+                    @Override
+                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                        if (Files.isHidden(dir) || ignoreFolders.contains(dir.getFileName().toString())) {
+                            return FileVisitResult.SKIP_SUBTREE;
+                        }
+                        return FileVisitResult.CONTINUE;
                     }
-                    return FileVisitResult.CONTINUE;
-                }
 
-                @Override
-                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
-                    String fileName = file.getFileName().toString();
-                    boolean shouldIgnore = Files.isHidden(file)
-                            || file.startsWith(outputDir)
-                            || ignoreFilePatterns.stream().anyMatch(p -> p.matcher(fileName).matches());
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        String fileName = file.getFileName().toString();
+                        boolean shouldIgnore = Files.isHidden(file)
+                                || file.startsWith(outputDir)
+                                || ignoreFilePatterns.stream().anyMatch(p -> p.matcher(fileName).matches());
 
-                    if (!shouldIgnore) {
-                        String name = projectRoot.relativize(file).toString();
-                        workspaceItems.add(new WorkspaceBuildItem.WorkspaceItem(name, file));
+                        if (!shouldIgnore) {
+                            String name = projectRoot.relativize(file).toString();
+                            workspaceItems.add(new WorkspaceBuildItem.WorkspaceItem(name, file));
+                        }
+                        return FileVisitResult.CONTINUE;
                     }
-                    return FileVisitResult.CONTINUE;
-                }
-            });
+                });
 
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+
+            sortWorkspaceItems(workspaceItems);
+            workspaceProducer.produce(new WorkspaceBuildItem(workspaceItems));
         }
-
-        sortWorkspaceItems(workspaceItems);
-        workspaceProducer.produce(new WorkspaceBuildItem(workspaceItems));
     }
 
     @BuildStep
-    InternalPageBuildItem createReportIssuePage() {
+    InternalPageBuildItem createWorkspacePage() {
         InternalPageBuildItem item = new InternalPageBuildItem("Workspace", 25);
 
         item.addPage(Page.webComponentPageBuilder().internal()
@@ -120,63 +129,66 @@ public class WorkspaceProcessor {
     }
 
     @BuildStep
-    void createBuildTimeActions(WorkspaceBuildItem workspaceBuildItem,
+    void createBuildTimeActions(Optional<WorkspaceBuildItem> workspaceBuildItem,
             List<WorkspaceActionBuildItem> workspaceActionBuildItems,
             BuildProducer<BuildTimeActionBuildItem> buildTimeActionProducer) {
 
-        // Dev UI Build Time Actions
-        BuildTimeActionBuildItem buildItemActions = new BuildTimeActionBuildItem(NAMESPACE);
+        if (workspaceBuildItem.isPresent()) {
 
-        // Workspace Actions
-        Map<String, Action> actionMap = workspaceActionBuildItems.stream()
-                .flatMap(item -> item.getActions().stream())
-                .map(ActionBuilder::build)
-                .collect(Collectors.toMap(Action::getId, action -> action, (a, b) -> a));
+            // Dev UI Build Time Actions
+            BuildTimeActionBuildItem buildItemActions = new BuildTimeActionBuildItem(NAMESPACE);
 
-        buildItemActions.addAction("getWorkspaceItems", (t) -> {
-            return workspaceBuildItem.getWorkspaceItems();
-        });
+            // Workspace Actions
+            Map<String, Action> actionMap = workspaceActionBuildItems.stream()
+                    .flatMap(item -> item.getActions().stream())
+                    .map(ActionBuilder::build)
+                    .collect(Collectors.toMap(Action::getId, action -> action, (a, b) -> a));
 
-        buildItemActions.addAction("getWorkspaceActions", (t) -> {
-            return actionMap.values().stream()
-                    .map(action -> new WorkspaceAction(action.getId(), action.getLabel(), action.getFilter(),
-                            action.getDisplay(), action.getDisplayType()))
-                    .sorted(Comparator.comparing(WorkspaceAction::label))
-                    .collect(Collectors.toList());
-        });
+            buildItemActions.addAction("getWorkspaceItems", (t) -> {
+                return workspaceBuildItem.get().getWorkspaceItems();
+            });
 
-        buildItemActions.addAction("executeAction", (t) -> {
+            buildItemActions.addAction("getWorkspaceActions", (t) -> {
+                return actionMap.values().stream()
+                        .map(action -> new WorkspaceAction(action.getId(), action.getLabel(), action.getFilter(),
+                                action.getDisplay(), action.getDisplayType()))
+                        .sorted(Comparator.comparing(WorkspaceAction::label))
+                        .collect(Collectors.toList());
+            });
 
-            String actionId = t.get("actionId");
+            buildItemActions.addAction("executeAction", (t) -> {
 
-            if (actionId != null) {
-                Path path = Path.of(URI.create(t.get("path")));
-                Action actionToExecute = actionMap.get(actionId);
-                Path convertedPath = (Path) actionToExecute.getPathConverter().apply(path);
-                return new WorkspaceActionResult(convertedPath, actionToExecute.getFunction().apply(t));
-            }
-            return null;
-        });
+                String actionId = t.get("actionId");
 
-        buildItemActions.addAction("getWorkspaceItemContent", (Map<String, String> params) -> {
-            if (params.containsKey("path")) {
-                Path path = Paths.get(URI.create(params.get("path")));
-                return readContents(path);
-            }
-            return null;
-        });
+                if (actionId != null) {
+                    Path path = Path.of(URI.create(t.get("path")));
+                    Action actionToExecute = actionMap.get(actionId);
+                    Path convertedPath = (Path) actionToExecute.getPathConverter().apply(path);
+                    return new WorkspaceActionResult(convertedPath, actionToExecute.getFunction().apply(t));
+                }
+                return null;
+            });
 
-        buildItemActions.addAction("saveWorkspaceItemContent", (Map<String, String> params) -> {
-            if (params.containsKey("content")) {
-                String content = params.get("content");
-                Path path = Paths.get(URI.create(params.get("path")));
-                writeContent(path, content);
-                return new SavedResult(path, true, null);
-            }
-            return new SavedResult(null, false, "Invalid input");
-        });
+            buildItemActions.addAction("getWorkspaceItemContent", (Map<String, String> params) -> {
+                if (params.containsKey("path")) {
+                    Path path = Paths.get(URI.create(params.get("path")));
+                    return readContents(path);
+                }
+                return null;
+            });
 
-        buildTimeActionProducer.produce(buildItemActions);
+            buildItemActions.addAction("saveWorkspaceItemContent", (Map<String, String> params) -> {
+                if (params.containsKey("content")) {
+                    String content = params.get("content");
+                    Path path = Paths.get(URI.create(params.get("path")));
+                    writeContent(path, content);
+                    return new SavedResult(path, true, null);
+                }
+                return new SavedResult(null, false, "Invalid input");
+            });
+
+            buildTimeActionProducer.produce(buildItemActions);
+        }
     }
 
     private void sortWorkspaceItems(List<WorkspaceBuildItem.WorkspaceItem> items) {

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java
@@ -11,8 +11,11 @@ import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
 import java.time.temporal.TemporalAccessor;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Locale;
 import java.util.Set;
+import java.util.stream.IntStream;
 
 import io.vertx.core.Handler;
 import io.vertx.core.buffer.Buffer;
@@ -44,7 +47,7 @@ public class MvnpmHandler implements Handler<RoutingContext> {
         }
 
         try {
-            URL url = mvnpmLoader.getResource(BASE_DIR + fullPath);
+            URL url = getUrl(fullPath);
             if (url != null) {
                 URLConnection openConnection = url.openConnection();
                 long lastModified = openConnection.getLastModified();
@@ -65,6 +68,35 @@ public class MvnpmHandler implements Handler<RoutingContext> {
             throw new UncheckedIOException(ex);
         }
         event.next();
+    }
+
+    private URL getUrl(String fullPath) {
+        URL url = mvnpmLoader.getResource(BASE_DIR + fullPath);
+        if (url != null)
+            return url;
+
+        // HACK: If url is null, it might be that the content is actually available under a package folder, ex
+        // /_static/lang-java/6.0.1/dist/index.js vs
+        // /_static/lang-java/6.0.1/package/dist/index.js
+        // This is due to a bug in mvnpm that at the time synced a few libraries packaged incorrectly
+        // This will eventually never run anymore, if newer versions of js packages in question updates
+        fullPath = insertPackageSegment(fullPath);
+        return mvnpmLoader.getResource(BASE_DIR + fullPath);
+    }
+
+    private String insertPackageSegment(String path) {
+        String[] parts = path.split(SLASH);
+        List<String> modifiedParts = new ArrayList<>();
+
+        IntStream.range(0, parts.length)
+                .forEach(i -> {
+                    modifiedParts.add(parts[i]);
+                    if (i == 3) { // After the version segment
+                        modifiedParts.add("package");
+                    }
+                });
+
+        return String.join(SLASH, modifiedParts);
     }
 
     private String formatDate(long m) {

